File: python/49_Group_Anagrams.py
class Group_Anagrams:
  def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
    myHash = dict()
    for word in strs:
      cur = ''.join(sorted(word))
      if cur in myHash:
        myHash[cur].append(word)
      else:
        myHash[cur] = [word]
    return myHash.values()

File: c++/49_Group_Anagrams.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

class GroupAnagrams {
public:
  vector<vector<string>> groupAnagrams(vector<string>& strs) {
    vector<vector<string>> result;
    unordered_map<string, vector<string>> hashmap;
    for (auto word:strs){
      string cur = word;
      sort(cur.begin(), cur.end());
      hashmap[cur].push_back(word);
    }
    for (auto x:hashmap){
      result.push_back(x.second);
    }
    return result;
  }
};